### PR TITLE
Support full-text index decorator in babel plugin

### LINKED ISSUE #5800: Update babel plugin with full text search support
After https://github.com/realm/realm-js/pull/5797 is merged, we should add support for the new index type with the babel plugin

### PR DESCRIPTION
**Pull request by atdyer**
This closes #5800 by adding support for the `index("full-text")` decorator.

- Check for `index` decorator calls that are passed the string "full-text" and subsequently add the `{ indexed: "full-text" }` property to the schema.
- Update README with description and example of indexing a string field by full text
- Update tests to verify schemas with full-text indexing are properly generated

### CODE REVIEW

**Review by kneth — APPROVED** [review]

**Comment by kneth** [comment]
@atdyer Thank you for your contribution. It looks good to me, and I have requested the team to review your PR.

I like that you have taken the time to update `README.md` and added tests.

Some test suites will fail since you don't have permission to launch some component - don't worry about that.

Once merged, we will update `CHANGELOG.md` and do a release of the babel plugin.

File: packages/babel-plugin/src/index.test.ts
Comment (takameyer) [reply packages/babel-plugin/src/index.test.ts:428]: I agree that an error should be thrown in this case.

File: packages/babel-plugin/src/plugin/index.ts
Comment (takameyer) [reply packages/babel-plugin/src/plugin/index.ts:297]: It should just forward whatever is given to the schema.  Then at least we get a run-time error.
Unfortunately it's not possible to get type errors here, which would be great.

**Review by takameyer — CHANGES_REQUESTED** [review]
I would approve this after the comments from @kraenhansen are resolved.

File: packages/babel-plugin/src/index.test.ts
Comment (elle-j) [reply packages/babel-plugin/src/index.test.ts:428]: Helping the user understand what's wrong and how to correct it also aligns better with the goals of the updated schema parser 👍

**Review by elle-j — COMMENTED** [review]
Thanks for taking the time to understand this issue and propose a solution! 🚀

**Comment by atdyer** [comment]
@takameyer I believe I've addressed all your comments - now any string passed into the `@index` decorator as a parameter will be passed through to the final schema. If there's anything I've missed please let me know an I'll fix it up!

**Review by kraenhansen — APPROVED** [review]
As far as I can tell, this just needs an entry in the [changelog](https://github.com/realm/realm-js/blob/main/packages/babel-plugin/CHANGELOG.md).

**Review by takameyer — APPROVED** [review]
Looks good to me 👍🏼

**Review by elle-j — APPROVED** [review]
Nice update!

**Pull request merged**
